File: 백준/Silver/28278. 스택 2/스택 2.py
import sys
input = sys.stdin.readline

n = int(input())
stack = []
for _ in range(n):
    num = list(map(int, input().split()))
    if num[0] == 1:    # push
        stack.append(num[1])
    elif num[0] == 2:  # 맨 위의 정수를 뺴고 출력
        if(len(stack) != 0):
            print(stack[len(stack)-1])
            stack.pop(len(stack)-1)
        else:
            print(-1)
    elif num[0] == 3:  # 정수의 개수 출력
        print(len(stack))
    elif num[0] == 4:  # isempty
        if len(stack) == 0:
            print(1)
        else:
            print(0)
    else:
        if(len(stack) != 0):
            print(stack[len(stack)-1])
        else:
            print(-1)

File: 백준/Silver/28278. 스택 2/스택 2.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        ArrayList<Integer> stack = new ArrayList<>();
        int n = Integer.parseInt(br.readLine());

        for (int i = 0; i < n; i++) {
            String inst = br.readLine();
            StringTokenizer st = new StringTokenizer(inst);

            while (st.hasMoreTokens()) {
                int value = Integer.parseInt(st.nextToken());

                switch (value) {
                    case 1:
                        push(stack, Integer.parseInt(st.nextToken()));
                        break;
                    case 2:
                        if(stack.isEmpty()){
                            System.out.println(-1);
                        }
                        else{
                            System.out.println(pop(stack));
                        }
                        break;
                    case 3:
                        System.out.println(size(stack));
                        break;
                    case 4:
                        System.out.println(isEmpty(stack));
                        break;
                    case 5:
                        if(stack.isEmpty()){
                            System.out.println(-1);
                        }
                        else {
                            System.out.println(printNum(stack));
                        }

                }
            }

        }

    }

    public static void push(ArrayList<Integer> arr, int num){
        arr.add(num);
    }

    public static int pop(ArrayList<Integer> arr){
        int num = arr.get(arr.size() - 1);
        arr.remove(arr.size() - 1);

        return num;
    }

    public static int size(ArrayList<Integer> arr){
        return arr.size();
    }

    public static int isEmpty(ArrayList<Integer> arr){
        int num = arr.size();

        if(num == 0){
            return 1;
        }
        else {
            return 0;
        }
    }

    public static int printNum(ArrayList<Integer> arr){
        return arr.get(arr.size() - 1);
    }
}
